utils/common/src/main/kotlin/DirectoryStash.kt

```diff
@@ -32,24 +32,27 @@ import kotlin.io.path.moveTo
 import org.apache.logging.log4j.kotlin.Logging
 
 /**
- * A convenience function that stashes directories using a [DirectoryStash] instance.
+ * A convenience function that stashes [directories] to the [stashParentDir] using a [DirectoryStash] instance.
  */
-fun stashDirectories(vararg directories: File): Closeable = DirectoryStash(setOf(*directories))
+fun stashDirectories(
+    vararg directories: File,
+    stashParentDir: File = createTempDirectory("stashParent").toFile()
+): Closeable = DirectoryStash(setOf(*directories), stashParentDir)
 
 /**
- * A Closable class which temporarily moves away directories and moves them back on close. Any conflicting directory
- * created at the location of an original directory is deleted before the original state is restored. If a specified
- * directory did not exist on initialization, it will also not exist on close.
+ * A [Closeable] class which temporarily moves away [directories] to inside a [stashParentDir] and moves them back on
```

Comment:
I believe the term `parent` is an implementation detail and should not be exposed, as this IMO leads to confusion. Would it make sense to rename this to `targetDir` or `stashDir` or `tempDir` ?

Reply:
> I believe the term parent is an implementation detail

In what way is it an implementation detail? It's the parent directory that holds all stashed directories, which IMO is something that the user should know in order to chose a proper location for the directory.

I'd rather say that whether `stashParentDir` is a *temporary* directory or not is something that's up to the user (now), so `tempDir` as a name would be misleading.

Reply:
I believe the reason the name contains `parent` is because the implementation creates another directory inside that to ensure it does not exist. That's what I meant with implementation detail.

Reply:
This became irrelevant due to the new approach.